models/models.js

```diff
@@ -1,8 +1,35 @@
 const db = require("../db/connection")
+const fs = require("fs/promises")
+const path = require("path");
+
 
 
 exports.selectTopics = () => {
     return db.query('SELECT * FROM topics;').then((topics) => {
         return topics.rows
     })
-}
\ No newline at end of file
+}
+
+exports.selectEndpoints = () => {
+    const filePath = path.resolve(__dirname, '../endpoints.json')
+    return fs.readFile(filePath, 'utf-8')
+     .then((data) => {
+        return data
+        
+    }).catch((err) => {
+        return err
+    })
+
+}
+
+exports.selectArticleByID = (id) => {
+    if (isNaN(Number(id))) return Promise.reject({status: 400, msg: 'Bad Request'})
```

Comment:
It's OK to handle it here, but psql will also handle it for you if the article_id is not the right datatype.